### PR TITLE
[AFD] ReceiveActivity: Don't return STATUS_FILE_CLOSED in case of FCB overread

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose
Fix FileZilla 3.8 `ECONNRESET` error (based on patch by [KRosUser](https://jira.reactos.org/secure/ViewProfile.jspa?name=KRosUser))

JIRA issue: [CORE-18328](https://jira.reactos.org/browse/CORE-18328)

### CODE REVIEW

**Comment by KRosUser2** [comment]
Lgtm, thanks !

**Comment by KRosUser2** [comment]
Tested 20+ times with Filezilla on a VM running this patch locally for 12+days with continuous network activity : 
- Problem is correctly fixed (see below)
- No adverse effect on other network related functions (browsing, SMTP, FTP, HTTP)
image

**Review by KRosUser2 — APPROVED** [review]

**Comment by ThFabba** [comment]
Do we have anything in our test suite that confirms that this change is correct? If not, it would be helpful to see an added test that we can refer back to to avoid regressions.

**Comment by KRosUser2** [comment]
In fact the question is :
- Is it appropriate to "force" the status to STATUS_FILE_CLOSED in case of OverRead when the FCB returns a succesful status ?
- Is it legit to raise an "Overread" situation in this case ?

This patch fixes the 1st point and makes FileZilla operative in ReactOS.

Regarding the test you need, I undestand your point as "we need test in testman" to check against possible future regressions, is it correct ?
@TAN-Gaming : Do you know how to do this ? (I don't)

**Comment by KRosUser2** [comment]
This may be a hack, but at least it enables FileZilla client in ReactOS and logs an ERR which would allow the "full fix" later if it's not.

**Comment by ThFabba** [comment]
Let's get a testman comparison of before and after please. Has anyone done a buildbot run against this?

**Comment by KRosUser2** [comment]
@ThFabba : results posted here : https://jira.reactos.org/browse/CORE-18328?focusedId=138889&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-138889 by @Doug-Lyons

File: drivers/network/afd/afd/read.c
Comment (ThFabba) [inline drivers/network/afd/afd/read.c:178]: This is the only use of the `Overread` member. Please remove the member entirely.

Comment (ThFabba) [inline drivers/network/afd/afd/read.c:178]: The new test in `ws2_32:recv` confirms that this special behavior is never needed, and this is the only use of the `Overread` member. Please remove it entirely.

File: drivers/network/afd/afd/read.c
Comment (ThFabba) [inline drivers/network/afd/afd/read.c:185]: No need for this comment, it's backed by tests (and we won't have to track "Overread" at all)

File: drivers/network/afd/afd/read.c
Comment (ThFabba) [inline drivers/network/afd/afd/read.c:187]: The `initial read after a` part is now superfluous -- graceful disconnect always returns success as shown by the test.

**Comment by KRosUser2** [comment]
Tested OK :
image

**Comment by atharva1910** [comment]
I agree with @KRosUser2, forcing a file close status seems strange.

I see you have also deleted the Overread flag. I have a question: 
Do we not care about over reading? If we are not returning an error does that not mean the client might potentially continue to read after EOF (the check above checks for EOF)? Would it not be apt to return a less harsh status? One which keeps does not force bugzilla (or any other application) to reset the connection?

**Comment by KRosUser2** [comment]
@ThFabba : Do you need anything more before approval ?

**Comment by atharva1910** [comment]
LGTM

**Review by atharva1910 — APPROVED** [review]

**Review by JoachimHenze — APPROVED** [review]

**Comment by Doug-Lyons** [comment]
Thanks @GeoB99 for the merge.

**Comment by KRosUser2** [comment]
Thanks @GeoB99 and @TAN-Gaming for this PR and its merge !

**Pull request merged**
